Explain the possible parameters when webpack.config.js exports a function

jennings opened the pull request:
This adds a table of possible `--env` arguments and how they translate into  the `env` parameter.

It also mentions the second parameter, `argv`, and gives an example of detecting if `-p` or `--optimize-minimize` was given.

skipjack commented on content/configuration/configuration-types.md:
```diff
@@ -11,21 +11,38 @@ contributors:
+| webpack --env.prod=1         | { prod: 1 }               |
+| webpack --env.prod=foo       | { prod: "foo" }           |
+| webpack --env.prod --env.bar | { prod: true, bar: true } |
+| webpack --env.prod --env bar | [{ prod: true }, "bar"]   |
```
The majority of this should be moved to the CLI section of the documentation, probably in [this section](https://webpack.js.org/api/cli/#common-options) as that's the only place I'm seeing `--env` mentioned. Then, from this page, we should link to there (feel free to create a dedicated section for `--env` with a header so you can link directly to it's hash).

jennings commented:
@skipjack Thanks for your help, I rebased and updated based on your suggestions, let me know if you'd like me to change anything else.

skipjack commented on content/api/cli.md:
```diff
@@ -129,6 +119,25 @@ webpack --json > stats.json
 
 In every other case, webpack prints out a set of stats showing bundle, chunk and timing details. Using this option the output can be a JSON object. This response is accepted by webpack's [analyse tool](https://webpack.github.com/analyse), or chrisbateman's [webpack-visualizer](https://chrisbateman.github.io/webpack-visualizer/), or th0r's [webpack-bundle-analyzer](https://github.com/th0r/webpack- ...
 
+### Environment options
```
I would capitalize "Options" as it's in the title.

skipjack commented on content/configuration/configuration-types.md:
```diff
@@ -11,21 +11,26 @@ contributors:
 
 
-## Exporting a function to use `--env`
+## Exporting a function to use `--env` and `argv`
```
Maybe simplify this title to "Exporting a Function"? Then in `api/cli.md` update your link to go directly here, e.g. `/configuration/configuration-types#exporting-a-function`.

skipjack commented on content/configuration/configuration-types.md:
```diff
@@ -11,21 +11,26 @@ contributors:
+One option is to export a function from your webpack config instead of exporting an object. The function will be invoked with two arguments:
+
+* An environment as the first parameter. See the [environment options CLI documentation](/api/cli#environment-options) for syntax examples.
+
```
Drop this line break... typically between list items we don't include them.

jennings commented:
@skipjack Updated:

* Capitalized "Environment Options"
* Removed the blank space between list items
* Rename the title to "Exporting a Function" and fixed up the link
* I also noticed I forgot to surround the environment examples with backticks, so I added those

skipjack commented:
@jennings thanks! Awesome work on this as well as #1489. If you're interested in contributing regularly, I would be happy to invite you to the organization. I feel like we're finally getting this repo into a good place but there is still a little ways to go with the backlog before we can fully focus on cleaning things up, reviewing the content, and organizing it all properly.

Anyways any time you can spare is definitely appreciated.

jennings commented:
@skipjack Thanks :)

I do want to contribute more. I'm not certain yet how much time I can devote to it, but, I can give it a shot. Anything specific you would want me to go after? I saw #1258, it seems like most topics in your checklist have been "claimed".

I'm fairly new to webpack, but I don't mind learning something new in order to improve the docs.

skipjack commented:
Yeah #1258 is pretty close to completion as you can see in [this comment](https://github.com/webpack/webpack.js.org/issues/1258#issuecomment-312139191). The ones left are somewhat advanced so I don't know if that would be a good start. 

If you could pick up any of the miscellaneous issues though that would be great and feel free to comment on them to discuss direction, e.g. where the new content might go if it's not obvious or just general thoughts on how the issue can be resolved.
